DOC-3134: Update documentation for Hyperlinking service using Docker (individually licensed) (#3629)
tinymce/tinymce-docs: modules/ROOT/partials/docker/hyperlinking-service/hyperlinking-service-installation.adoc

[[installation]]
== Installation

[NOTE]

> A valid license key is needed in order to install {linkcheckerplugin} and {mediaembedplugin} On-Premises.

Reviewer: Verify this is the same key, if so we should link to this page.
```suggestion
A valid xref:license-key.adoc[License Key] is **required** in order to install {linkcheckerplugin} and {mediaembedplugin} On-Premises.
```

Author: Yes, it's referencing the same key.

> link:https://www.tiny.cloud/contact/[Contact us] for a trial license key.

Reviewer: ```suggestion
link:https://www.tiny.cloud/contact/[Contact us] for a trial xref:license-key.adoc[License Key].
```

=== Retrieve Docker Image

> . Login into the Tiny Cloud Docker Registry: 

Reviewer: ```suggestion
. Login into the {companyname} Cloud Docker Registry: 
```

+
[source, sh, subs="attributes+"]
----
docker login -u [username] -p [password] registry.containers.tiny.cloud
----

. Pull the Docker Image from the Docker registry:
+
[source, sh, subs="attributes+"]
----
docker pull registry.containers.tiny.cloud/hyperlinking-tiny:<VERSION>
----
+

> Replace `<VERSION>` with `latest` or the specific version number you’re looking for (eg: `2.109.0`)

Reviewer: Simplify this, question if we mention the specific version num, do we have a list of available versions?
```suggestion
Replace `<VERSION>` with `latest` or the specific version number.
```

Author: I'm not sure. I'll check if we have a list to reference.

2025-01-02 11:06:46 [io-compute-9] INFO  o.h.b.c.nio1.NIO1SocketServerGroup - Service bound to address /0:0:0:0:0:0:0:0:19100
2025-01-02 11:06:46 [io-compute-9] INFO  o.h.blaze.server.BlazeServerBuilder - 
  _   _   _        _ _
 | |_| |_| |_ _ __| | | ___
 | ' \\  _|  _| '_ \\_  _(_-<
 |_||_\\__|\\__| .__/ |_|/__/
             |_|
2025-01-02 11:06:46 [io-compute-9] INFO  o.h.blaze.server.BlazeServerBuilder - http4s v0.23.27 on blaze v0.23.16 started at http://[::]:19100/
----

Running this command will generate a log warning about `allowed-origins` not being configured. This is expected, as it will be set up in the next step.

The {productname} server-side components require a configuration file to function correctly. By convention, this file is named `application.conf`. For more information, refer to link:https://www.tiny.cloud/docs/tinymce/latest/configure-required-services/[Required configuration for the server-side components]. 

This configuration file requires at least the following information:

> * allowed-origins - Specifies the domains allowed to communicate with server-side editor features. This is mandatory for all server-side components.

Reviewer: ```suggestion
* allowed-origins - Specifies the domains allowed to communicate with server-side editor features. This is **mandatory** for all server-side components.
```

The following settings for the premium server-side components are optional and will apply to all services that make outgoing HTTP/HTTPS requests using the configuration file.

* link:https://www.tiny.cloud/docs/tinymce/latest/configure-common-settings-services/#proxy[proxy configuration]
* link:https://www.tiny.cloud/docs/tinymce/latest/configure-common-settings-services/#http[http configuration]

> The following settings for the Hyperlinking premium server-side component are optional:

Reviewer: ```suggestion
The following settings for the Hyperlinking premium server-side component are _optional_:
```

* link:https://www.tiny.cloud/docs/tinymce/latest/configure-hyperlink-service/#link-checking[link-checking configuration]
* link:https://www.tiny.cloud/docs/tinymce/latest/configure-hyperlink-service/#cache[cache configuration]

> The {mediaembedplugin} service allows the use of a custom Iframely account, configured oEmbed endpoints, or a combination of both. If a URL lacks an Iframely or oEmbed configuration, the service generates a summary card.

Reviewer: Add links to Iframely and oEmbed

This configuration file requires at least the following information:

* allowed-origins - Specifies the domains allowed to communicate with server-side editor features. This is **mandatory** for all server-side components.

The following settings for the premium server-side components are _optional_ and will apply to all services that make outgoing HTTP/HTTPS requests using the configuration file.

* link:https://www.tiny.cloud/docs/tinymce/latest/configure-common-settings-services/#proxy[proxy configuration^]
* link:https://www.tiny.cloud/docs/tinymce/latest/configure-common-settings-services/#http[http configuration^]

The following settings for the Hyperlinking premium server-side component are _optional_:

* link:https://www.tiny.cloud/docs/tinymce/latest/configure-hyperlink-service/#link-checking[link-checking configuration^]
* link:https://www.tiny.cloud/docs/tinymce/latest/configure-hyperlink-service/#cache[cache configuration^]

> The {mediaembedplugin} service allows the use of a custom xref:mediaembed-server-config.adoc#use-your-own-iframely-account[Iframely^] account, configured xref:mediaembed-server-config.adoc#configure-a-custom-endpoint[oEmbed^] endpoints, or a combination of both. If a URL lacks an `Iframely` or `oEmbed` configuration, the service generates a summary card.

Reviewer: ```suggestion
The {mediaembedplugin} service allows the use of a custom xref:mediaembed-server-config.adoc#use-your-own-iframely-account[Iframely^] account, configured xref:mediaembed-server-config.adoc#configure-a-custom-endpoint[oEmbed^] endpoints, or a combination of both. If a URL lacks an `Iframely` or `oEmbed` configuration, the service generates a summary card.
```

[[installation]]
== Installation

include::partial$misc/admon-docker-access.adoc[]

=== Retrieve Docker Image

. Login into the {companyname} Cloud Docker Registry: 
+
[source, sh, subs="attributes+"]
----
docker login -u tiny -p [access-token] registry.containers.tiny.cloud
----

. Pull the Docker Image from the Docker registry:
+
[source, sh, subs="attributes+"]
----
docker pull registry.containers.tiny.cloud/hyperlinking-tiny:<VERSION>
----
+

> Replace `<VERSION>` with `latest` or the specific version number.
> 

Reviewer: ```suggestion
Replace `<VERSION>` with `latest` or the specific version number.

[NOTE]
Currently, the Docker images are only supported on x86-64 (also known as AMD64) architecture processors.
```

This configuration file requires at least the following information:

* allowed-origins - Specifies the domains allowed to communicate with server-side editor features. This is **mandatory** for all server-side components.

The following settings for the premium server-side components are _optional_ and will apply to all services that make outgoing HTTP/HTTPS requests using the configuration file.

* link:https://www.tiny.cloud/docs/tinymce/latest/configure-common-settings-services/#proxy[proxy configuration^]
* link:https://www.tiny.cloud/docs/tinymce/latest/configure-common-settings-services/#http[http configuration^]

The following settings for the Hyperlinking premium server-side component are _optional_:

* link:https://www.tiny.cloud/docs/tinymce/latest/configure-hyperlink-service/#link-checking[link-checking configuration^]
* link:https://www.tiny.cloud/docs/tinymce/latest/configure-hyperlink-service/#cache[cache configuration^]

> The {mediaembedplugin} service allows the use of a custom link:https://www.tiny.cloud/docs/tinymce/latest/mediaembed-server-config/#use-your-own-iframely-account[Iframely^] account, configured link:https://www.tiny.cloud/docs/tinymce/latest/mediaembed-server-config/#configure-a-custom-endpoint[oEmbed^] endpoints, or a combination of both. If a URL lacks an Iframely or oEmbed configuration, the service generates a summary card.

Reviewer: Verify the updated `xref` links work once it rebuilds staging. When we have links internal to our project we use `xref` links instead of external `link`'s
```suggestion
The {mediaembedplugin} service allows the use of a custom xref:mediaembed-server-config.adoc#use-your-own-iframely-account[Iframely^] account, configured xref:mediaembed-server-config.adoc#configure-a-custom-endpoint[oEmbed^] endpoints, or a combination of both. If a URL lacks an `Iframely` or `oEmbed` configuration, the service generates a summary card.
```

Author: @kemister85 Noted, thanks!

[[installation]]
== Installation

include::partial$misc/admon-docker-access.adoc[]

=== Retrieve Docker Image

. Login into the {companyname} Cloud Docker Registry: 
+
[source, sh, subs="attributes+"]
----
docker login -u tiny -p [access-token] registry.containers.tiny.cloud
----

. Pull the Docker Image from the Docker registry:
+
[source, sh, subs="attributes+"]
----
docker pull registry.containers.tiny.cloud/hyperlinking-tiny:<VERSION>
----
+
Replace `<VERSION>` with `latest` or the specific version number.

> [NOTE]
> Currently, the Docker images are only supported on x86-64 (also known as AMD64) architecture processors.
> +

Reviewer: Remove duplication as its now in partial
```suggestion
```
